server/game/cards/13-CW/RampartsOfStone.js

```diff
@@ -0,0 +1,36 @@
+const DrawCard = require('../../drawcard.js');
+const { Players, TargetModes, CardTypes } = require('../../Constants');
+const AbilityDsl = require('../../abilitydsl');
+
+class RampartsOfStone extends DrawCard {
+    setupCardAbilities() {
+        this.action({
+            title: 'Attacker bows participating characters or discards three cards from hand',
+            condition: context => context.game.isDuringConflict(),
+            targets: {
+                select: {
+                    mode: TargetModes.Select,
+                    player: (context) => {
+                        if(context.player.isAttackingPlayer()) {
+                            return Players.Self;
+                        }
+                        return Players.Opponent;
+                    },
+                    choices: {
+                        'Bow all participating characters': AbilityDsl.actions.bow((context) => {
+                            let targetPlayer = context.player.isAttackingPlayer() ? context.player : context.player.opponent;
+                            return {
+                                target: targetPlayer.filterCardsInPlay(card => card.getType() === CardTypes.Character && card.isParticipating())
```

Comment:
conflict has the method `getCharacters()` which accepts a player and returns their participating characters